rules/experimental/rbe/exec_properties.bzl

```diff
@@ -156,3 +247,108 @@ def merge_dicts(*dict_args):
+    implementation = _exec_property_sets_repository_impl,
+    local = True,
+    attrs = {
+        "exec_property_sets_content": attr.string(
+            mandatory = True,
+            doc = "The content of the exec_propert_sets.bzl file within the repository rule.",
+        ),
+    },
+)
+
+def _verify_dict_of_dicts(name, dicts):
+    """ Verify that dict is of type {string->{string->string}}.
+
+    Args:
+      name: Name of the repo rule. Used for error messages.
+      dicts: a dict whose key is a string and whose value is a dict from string to string.
+    """
+
+    # Verify that dict is of type {string->{string->string}}.
+    for key, value in dicts.items():
+        if type(key) != "string":
+            fail("In repo rule %s, execution property set name %s must be a string" % (name, key))
+        if type(value) != "dict":
+            fail("In repo rule %s, execution property set of %s must be a dict" % (name, key))
+        for k, v in value.items():
+            if type(k) != "string":
+                fail("In repo rule %s, execution property set %s, the key %s must be a string" % (name, key, k))
+            if type(v) != "string":
+                fail("In repo rule %s, execution property set %s, key %s, the value %s must be a string" % (name, key, k, v))
+
+def custom_exec_properties(name, dicts):
+    """ Creates a repository containing execution property dicts.
+
+    Use this macro in your WORKSPACE.
+
+    Args:
+      name: Name of the repo rule.
+      dicts: The set of execution property sets.
+    """
+    _verify_dict_of_dicts(name, dicts)
+
+    exec_property_sets_content = ""
+    for key, value in dicts.items():
+        exec_property_sets_content += "%s = %s\n" % (key, value)
+
+    _exec_property_sets_repository(
+        name = name,
+        exec_property_sets_content = exec_property_sets_content,
+    )
+
+STANDARD_PROPERTY_SETS = {
+    "NETWORK_ON": create_exec_properties_dict(docker_network = "standard"),
+    "NETWORK_OFF": create_exec_properties_dict(docker_network = "off"),
+    "DOCKER_PRIVILEGED": create_exec_properties_dict(docker_privileged = True),
+    "NOT_DOCKER_PRIVILEGED": create_exec_properties_dict(docker_privileged = False),
+    "DOCKER_RUN_AS_ROOT": create_exec_properties_dict(docker_run_as_root = True),
+    "NOT_DOCKER_RUN_AS_ROOT": create_exec_properties_dict(docker_run_as_root = False),
+    "DOCKER_SIBLINGS_CONTAINERS": create_exec_properties_dict(docker_sibling_containers = True),
+    "NOT_DOCKER_SIBLINGS_CONTAINERS": create_exec_properties_dict(docker_sibling_containers = False),
+    "DOCKER_USE_URANDOM": create_exec_properties_dict(docker_use_urandom = True),
+    "NOT_DOCKER_USE_URANDOM": create_exec_properties_dict(docker_use_urandom = False),
+    "LINUX": create_exec_properties_dict(os_family = "Linux"),
+    "WINDOWS": create_exec_properties_dict(os_family = "Windows"),
+    "NVIDIA": create_exec_properties_dict(docker_runtime = "nvidia"),
```

Comment:
Should we remove this from here?
I'll follow up with a doc, but basically: I don't think a user would ever want to use this standalone, currently they need this together with some way to identify the pool.
Let's talk more

Reply:
Done.